### DISCUSSION TITLE
context.imageLoader.diskCache?.get() is null

### DISCUSSION THREAD

**Discussion by HehuanAndroid** [coil-kt/coil#1364] [General]
when i  do   

imageLoader.diskCache?.clear()
imageLoader.memoryCache?.clear()

uri.getCoilCacheFile()  is null

val imageLoader = getImageLoader(context)
        val request = ImageRequest.Builder(context)
            .data(uri)
            .crossfade(1000)  //淡入淡出时间
            .placeholder(R.drawable.ps_image_placeholder)
            .memoryCacheKey(uri.toString())
            .target(object : ImageDownloadTarget(uri?.toString().orEmpty()) {
                override fun onSuccess(result: Drawable) {
                    super.onSuccess(result)
                    callback?.onDrawSuccess(result)
                    var resource = uri.getCoilCacheFile()
                        if (resource?.exists() == true) {
                            callback?.onSuccess(resource)
                        } else {
                            callback?.onFail(CoilLoaderException(null))
                        }
                    }

                override fun onError(error: Drawable?) {
                    super.onError(error)
                    callback?.onFail(CoilLoaderException(error))
                }

                override fun onDownloadStart() {
                    callback?.onStart()
                }

                override fun onProgress(progress: Int) {
                    callback?.onProgress(progress)
                }

                override fun onDownloadFinish() {
                    callback?.onFinish()
                }

            })
            .build()

fun String?.getCoilCacheFile(): File? {
        return context.imageLoader.diskCache?.get(this.orEmpty())?.data?.toFile()
    }

**Comment by Nashmi97** [comment]
> عندما أفعل
> 
> imageLoader.diskCache؟. clear () imageLoader.memoryCache؟. clear ()
> 
> uri.getCoilCacheFile () فارغ
> 
> val imageLoader = getImageLoader (سياق) طلب فال = ImageRequest.Builder (سياق) . بيانات (uri) .crossfade (1000) // 淡入 淡出 时间 .placeholder (R.drawable.ps_image_placeholder) .memoryCacheKey (uri.toString ()) . target (object: ImageDownloadTarget (uri؟ .toString (). orEmpty ()) { override fun onSuccess (result: Drawable) { super.onSuccess (result) callback؟ .onDrawSuccess (result) var Resource = uri.getCoilCacheFile () if ( Resource؟ .exists () == true) { callback؟ .onSuccess (Resource) } else { callback؟ .onFail (CoilLoaderException (خالية)) } }
> 
> ```
>             override fun onError(error: Drawable?) {
>                 super.onError(error)
>                 callback?.onFail(CoilLoaderException(error))
>             }
> 
>             override fun onDownloadStart() {
>                 callback?.onStart()
>             }
> 
>             override fun onProgress(progress: Int) {
>                 callback?.onProgress(progress)
>             }
> 
>             override fun onDownloadFinish() {
>                 callback?.onFinish()
>             }
> 
>         })
>         .build()
> ```
> 
> fun String؟ .getCoilCacheFile (): File؟ { return context.imageLoader.diskCache؟ .get (this.orEmpty ()) ؟. data؟ .toFile () }

**Comment by Nashmi97** [comment]
https://github.com/coil-kt/coil/discussions/1364#discussion-4234834
